build.rs

```diff
@@ -8,33 +8,24 @@ use gl_generator::{Registry, Api, Profile, Fallbacks};
 
 fn main() {
     let dest = env::var("OUT_DIR").unwrap();
-    let mut file = File::create(&Path::new(&dest).join("gl_bindings.rs")).unwrap();
+    let mut file_gl_and_gles = File::create(&Path::new(&dest).join("gl_and_gles_bindings.rs")).unwrap();
+    let mut file_gl = File::create(&Path::new(&dest).join("gl_bindings.rs")).unwrap();
+    let mut file_gles = File::create(&Path::new(&dest).join("gles_bindings.rs")).unwrap();
 
-    let target = env::var("TARGET").unwrap();
+    // OpenGL 3.3 bindings
+    let gl_extensions = ["GL_ARB_texture_rectangle", "GL_EXT_debug_marker"];
+    let gl_reg = Registry::new(Api::Gl, (3, 3), Profile::Core, Fallbacks::All, gl_extensions);
+    gl_reg.write_bindings(gl_generator::StructGenerator, &mut file_gl)
+          .unwrap();
 
-    if target.contains("android") {
-        let extensions = ["GL_EXT_texture_format_BGRA8888"];
-        // GLES 2.0 bindings for Android
-        Registry::new(Api::Gles2, (3, 0), Profile::Core, Fallbacks::All, extensions)
-            .write_bindings(gl_generator::StaticGenerator, &mut file)
+    // GLES 2.0 bindings
+    let gles_extensions = ["GL_EXT_texture_format_BGRA8888"];
+    let gles_reg = Registry::new(Api::Gles2, (3, 0), Profile::Core, Fallbacks::All, gles_extensions);
+    gles_reg.write_bindings(gl_generator::StructGenerator, &mut file_gles)
             .unwrap();
 
-        println!("cargo:rustc-link-lib=GLESv3");
```

Comment:
> Is your idea to move that responsibility higher up to WR/Servo/Gecko?

We move to run time selection. My understanding to it is move to dynamic library loading like gecko. Is my understanding correct? Gecko does not do static link to EGL on android.